tests/utils_test.py

```diff
@@ -375,6 +375,23 @@ def test_fill_gaps_with_value(self):
 
         self.assertEqual(actual, expected)
 
+    def test_fill_gaps_with_non_square_array(self):
+        """test fill_gaps function"""
+        matrix = [
+            [1, 2, 3, 4],
+            [5, 6, 7, 8],
```

Comment:
Can you may be shift the second row `[5, 6, 7, 8]` to one on the right so we have a empty celle on the left side at the beginning of the line ?

`[None, 5, 6, 7 ,8]`

Reply:
I can do that, but `fill_gaps` doesn't ever *replace* values, it only ever adds new values to the arrays (either to make an array square, or to extend the rows/columns)

so, `fill_gaps` applied to `[None, 5, 6, 7, 8]` --> `[None, 5, 6, 7, 8]`.

This becomes a question of what `filil_gaps` *should* do in this situation. It is just a `utils` function, so its scope was never fully defined. What do you think?

Reply:
I see then leave it as is.